atomicapp/providers/openshift.py

```diff
@@ -655,3 +757,93 @@ def _set_config_values(self):
                                             result[PROVIDER_CA_KEY].lstrip('/'))
         else:
             self.provider_ca = None
+
+    def extract(self, image, src, dest, update=True):
+        """
+        Extract contents of a container image from 'src' in container
+        to 'dest' in host.
+
+        Args:
+            image (str): Name of container image
+            src (str): Source path in container
+            dest (str): Destination path in host
+            update (bool): Update existing destination, if True
+        """
+        if os.path.exists(dest) and not update:
+            return
+        cleaned_image_name = Utils.sanitizeName(image)
+        pod_name = '{}-{}'.format(cleaned_image_name, Utils.getUniqueUUID())
+        container_name = cleaned_image_name
+
+        # Pull (if needed) image and bring up a container from it
+        # with 'sleep 3600' entrypoint, just to extract content from it
+        artifact = {
+            'apiVersion': 'v1',
+            'kind': 'Pod',
+            'metadata': {
+                'name': pod_name
+            },
+            'spec': {
+                'containers': [
+                    {
+                        'image': image,
+                        'command': [
+                            'sleep',
+                            '3600'
+                        ],
+                        'imagePullPolicy': 'IfNotPresent',
+                        'name': container_name
+                    }
+                ],
+                'restartPolicy': 'Always'
+            }
+        }
+
+        self.oc.deploy(self._get_url(self.namespace, 'Pod'), artifact)
+        self._wait_till_pod_runs(self.namespace, pod_name, timeout=300)
+
+        # Archive content from the container and dump it to tmpfile
+        tmpfile = '/tmp/atomicapp-{pod}.tar.gz'.format(pod=pod_name)
+        self.oc.execute(
+            self.namespace, pod_name, container_name,
+            'tar -cz --directory {} ./'.format('/' + src),
+            outfile=tmpfile
+        )
```

Comment:
We need to put `_wait_till_pod_runs` and `oc.execute` inside of a try/finally where the finally contains the `oc.delete` of the pod.